README.md
# wazo-c4-router

wazo-c4-router provides kamailio configuration for ROUTER component of the C4 infrastructure.

## Installing wazo-c4-router

The server is already provided as a part of [Wazo](http://documentation.wazo.community).
Please refer to [the documentation](http://documentation.wazo.community/en/stable/installation/installsystem.html) for
further details on installing one.



debian/changelog
wazo-c4-router (1.0.1) wazo-dev-buster; urgency=medium

  * First package

 -- Wazo Maintainers <dev+pkg@wazo.community>  Tue, 23 Oct 2018 10:52:32 -0400

wazo-c4-sbc (1.0.1) wazo-dev-buster; urgency=medium

 * First package

-- Wazo Maintainers <dev+pkg@wazo.community>  Tue, 30 Oct 2019 10:52:32 -0400

wazo-c4-aio (1.0.1) wazo-dev-buster; urgency=medium

 * First package

-- Wazo Maintainers <dev+pkg@wazo.community>  Tue, 23 Mar 2020 10:52:32 -0400


docker-entrypoint-aio.sh
#!/bin/sh
date

# Wait For
if ! [ -z "$ROUTER_CONFD_URI" ]; then
    wait-for -t 60 $ROUTER_CONFD_URI
fi
if ! [ -z "$CONSUL_URI" ]; then
    wait-for -t 60 $CONSUL_URI
    sleep 2
fi
if ! [ -z "$REDIS_URI" ]; then
    wait-for -t 60 $REDIS_URI
    sleep 2
fi

# Interfaces and IPs
if [ -z "$INTERFACE_SIP" ]; then
    INTERFACE_SIP="eth0"
fi
if [ -z "$DISPATCHER_WEIGHT" ]; then
    DISPATCHER_WEIGHT="5"
fi
if [ "$DISPATCHER_WEIGHT" -gt "5" ]; then
    DISPATCHER_ORDER="primary"
else
    DISPATCHER_ORDER="secondary"
fi
if [ -z "$INTERFACE_DMQ" ]; then
    INTERFACE_DMQ="$INTERFACE_SIP"
fi
if [ -z "$INTERFACE_XHTTP" ]; then
    INTERFACE_XHTTP="$INTERFACE_SIP"
fi
SIP_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_SIP'" { gsub(/\/.*/, "", $4); print $4 }')
DMQ_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_DMQ'" { gsub(/\/.*/, "", $4); print $4 }')
XHTTP_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_XHTTP'" { gsub(/\/.*/, "", $4); print $4 }')

if [ -z "$XHTTP_PORT" ]; then
    XHTTP_PORT="9600"
fi

HOSTNAME=$(hostname)
export KAMAILIO=$(which kamailio)

# Kamailio-local.cfg
mkdir -p /etc/kamailio/ /etc/kamailio/dbtext

#statements
echo '#!define LISTEN_XHTTP tcp:'$INTERFACE_XHTTP':'$XHTTP_PORT > /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_ROUTING_ENDPOINT "'$HTTP_API_ROUTING_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_CDR_ENDPOINT "'$HTTP_API_CDR_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_DBTEXT_UACREG_ENDPOINT "'$HTTP_API_DBTEXT_UACREG_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_TIMEOUT '$HTTP_API_TIMEOUT >> /etc/kamailio/kamailio-local.cfg
echo '#!define LISTEN '$LISTEN >> /etc/kamailio/kamailio-local.cfg
if ! [ -z "$TESTING" ]; then
    echo '#!define TESTING 1' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$DISPATCHER_ALG" ]; then
    echo '#!define DISPATCHER_ALG "'$DISPATCHER_ALG'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$HTTP_ASYNC_CLIENT_WORKERS" ]; then
    echo '#!define HTTP_ASYNC_CLIENT_WORKERS '$HTTP_ASYNC_CLIENT_WORKERS  >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$LISTEN_ADVERTISE" ]; then
    echo '#!define LISTEN_ADVERTISE '$LISTEN_ADVERTISE >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$ALIAS" ]; then
    echo '#!define ALIAS '$ALIAS >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$WITH_DMQ" ]; then
    echo '#!define WITH_DMQ 1' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_PORT "'$DMQ_PORT'"' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_LISTEN '$DMQ_LISTEN >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_SERVER_ADDRESS "sip:'$DMQ_IP':'$DMQ_PORT'"' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_NOTIFICATION_ADDRESS "'$DMQ_NOTIFICATION_ADDRESS'"' >> /etc/kamailio/kamailio-local.cfg
    if ! [ -z "$DMQ_PING_INTERVAL" ]; then
     echo '#!define DMQ_PING_INTERVAL "'$DMQ_PING_INTERVAL'"' >> /etc/kamailio/kamailio-local.cfg
    fi
fi
if ! [ -z "$WITH_REDIS_DIALOG" ]; then
    echo '#!define WITH_REDIS_DIALOG 1' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$DBURL_DIALOG" ]; then
    echo '#!define DBURL_DIALOG "'$DBURL_DIALOG'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$NAT_PING_FROM" ]; then
    echo '#!define NAT_PING_FROM "'$NAT_PING_FROM'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$ROUTER_AUTH_SECRET" ]; then
    echo '#!define ROUTER_AUTH_SECRET "'$ROUTER_AUTH_SECRET'"' >> /etc/kamailio/kamailio-local.cfg
fi

# test the config syntax
$KAMAILIO -f $KAMAILIO_CONF -c

# register/de-register service in consul
curl -i -X PUT http://${CONSUL_URI}/v1/agent/service/register -d '{
    "ID": "'$HOSTNAME'",
    "Name": "sbc",
    "Tags": ["sbc", "kamailio", "'$DISPATCHER_ORDER'"],
    "Address": "'$SIP_IP'",
    "Port": '$SIP_PORT',
    "Check": {
        "ID": "XHTTP",
        "Name": "XHTTP API on port 9600",
        "DeregisterCriticalServiceAfter": "10m",
        "Method": "GET",
        "HTTP": "http://'$XHTTP_IP':'$XHTTP_PORT'/status",
        "Timeout": "1s",
        "Interval": "10s"
    },
    "Weights": {
      "Passing": '$DISPATCHER_WEIGHT',
      "Warning": 1
    }
}'

exit_script() {
    curl -X PUT http://${CONSUL_URI}/v1/agent/service/deregister/$HOSTNAME
    [ -f /var/run/supervisor.sock ] && supervisorctl -c /etc/supervisor/conf.d/supervisord.conf shutdown
    date
    exit 143; # 128 + 15 -- SIGTERM
}
trap exit_script SIGINT SIGTERM

# run through supervisor
supervisord=$(which supervisord)
$supervisord -n -c /etc/supervisor/conf.d/supervisord.conf &

# wait for signals
while true; do sleep 1; done

# exit
exit_script


docker-entrypoint-sbc.sh
#!/bin/sh
date

# Wait For
if ! [ -z "$ROUTER_CONFD_URI" ]; then
    wait-for -t 60 $ROUTER_CONFD_URI
fi
if ! [ -z "$CONSUL_URI" ]; then
    wait-for -t 60 $CONSUL_URI
    sleep 2
fi
if ! [ -z "$REDIS_URI" ]; then
    wait-for -t 60 $REDIS_URI
    sleep 2
fi

# Interfaces and IPs
if [ -z "$INTERFACE_SIP" ]; then
    INTERFACE_SIP="eth0"
fi
if [ -z "$DISPATCHER_WEIGHT" ]; then
    DISPATCHER_WEIGHT="5"
fi
if [ "$DISPATCHER_WEIGHT" -gt "5" ]; then
    DISPATCHER_ORDER="primary"
else
    DISPATCHER_ORDER="secondary"
fi
if [ -z "$INTERFACE_DMQ" ]; then
    INTERFACE_DMQ="$INTERFACE_SIP"
fi
if [ -z "$INTERFACE_XHTTP" ]; then
    INTERFACE_XHTTP="$INTERFACE_SIP"
fi
SIP_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_SIP'" { gsub(/\/.*/, "", $4); print $4 }')
DMQ_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_DMQ'" { gsub(/\/.*/, "", $4); print $4 }')
XHTTP_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_XHTTP'" { gsub(/\/.*/, "", $4); print $4 }')

if [ -z "$XHTTP_PORT" ]; then
    XHTTP_PORT="9600"
fi

HOSTNAME=$(hostname)
export KAMAILIO=$(which kamailio)

# Kamailio-local.cfg
mkdir -p /etc/kamailio/ /etc/kamailio/dbtext

# If SBC Role
echo '#!define IS_SBC' > /etc/kamailio/kamailio-local.cfg

#statements
echo '#!define LISTEN_XHTTP tcp:'$INTERFACE_XHTTP':'$XHTTP_PORT > /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_ROUTING_ENDPOINT "'$HTTP_API_ROUTING_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_CDR_ENDPOINT "'$HTTP_API_CDR_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_DBTEXT_UACREG_ENDPOINT "'$HTTP_API_DBTEXT_UACREG_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_TIMEOUT '$HTTP_API_TIMEOUT >> /etc/kamailio/kamailio-local.cfg
echo '#!define LISTEN '$LISTEN >> /etc/kamailio/kamailio-local.cfg
if ! [ -z "$TESTING" ]; then
    echo '#!define TESTING 1' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$DISPATCHER_ALG" ]; then
    echo '#!define DISPATCHER_ALG "'$DISPATCHER_ALG'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$HTTP_ASYNC_CLIENT_WORKERS" ]; then
    echo '#!define HTTP_ASYNC_CLIENT_WORKERS '$HTTP_ASYNC_CLIENT_WORKERS  >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$LISTEN_ADVERTISE" ]; then
    echo '#!define LISTEN_ADVERTISE '$LISTEN_ADVERTISE >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$ALIAS" ]; then
    echo '#!define ALIAS '$ALIAS >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$WITH_DMQ" ]; then
    echo '#!define WITH_DMQ 1' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_PORT "'$DMQ_PORT'"' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_LISTEN '$DMQ_LISTEN >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_SERVER_ADDRESS "sip:'$DMQ_IP':'$DMQ_PORT'"' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_NOTIFICATION_ADDRESS "'$DMQ_NOTIFICATION_ADDRESS'"' >> /etc/kamailio/kamailio-local.cfg
    if ! [ -z "$DMQ_PING_INTERVAL" ]; then
     echo '#!define DMQ_PING_INTERVAL "'$DMQ_PING_INTERVAL'"' >> /etc/kamailio/kamailio-local.cfg
    fi
fi
if ! [ -z "$WITH_REDIS_DIALOG" ]; then
    echo '#!define WITH_REDIS_DIALOG 1' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$DBURL_DIALOG" ]; then
    echo '#!define DBURL_DIALOG "'$DBURL_DIALOG'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$NAT_PING_FROM" ]; then
    echo '#!define NAT_PING_FROM "'$NAT_PING_FROM'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$ROUTER_AUTH_SECRET" ]; then
    echo '#!define ROUTER_AUTH_SECRET "'$ROUTER_AUTH_SECRET'"' >> /etc/kamailio/kamailio-local.cfg
fi

# test the config syntax
$KAMAILIO -f $KAMAILIO_CONF -c

# register/de-register service in consul
curl -i -X PUT http://${CONSUL_URI}/v1/agent/service/register -d '{
    "ID": "'$HOSTNAME'",
    "Name": "sbc",
    "Tags": ["sbc", "kamailio"],
    "Address": "'$SIP_IP'",
    "Port": '$SIP_PORT',
    "Check": {
        "ID": "XHTTP",
        "Name": "XHTTP API on port 9600",
        "DeregisterCriticalServiceAfter": "10m",
        "Method": "GET",
        "HTTP": "http://'$XHTTP_IP':'$XHTTP_PORT'/status",
        "Timeout": "1s",
        "Interval": "10s"
    }
}'

exit_script() {
    curl -X PUT http://${CONSUL_URI}/v1/agent/service/deregister/$HOSTNAME
    [ -f /var/run/supervisor.sock ] && supervisorctl -c /etc/supervisor/conf.d/supervisord.conf shutdown
    date
    exit 143; # 128 + 15 -- SIGTERM
}
trap exit_script SIGINT SIGTERM

# run through supervisor
supervisord=$(which supervisord)
$supervisord -n -c /etc/supervisor/conf.d/supervisord.conf &

# wait for signals
while true; do sleep 1; done

# exit
exit_script


docker-entrypoint-router.sh
#!/bin/sh
date

# Wait For
if ! [ -z "$ROUTER_CONFD_URI" ]; then
    wait-for -t 60 $ROUTER_CONFD_URI
fi
if ! [ -z "$CONSUL_URI" ]; then
    wait-for -t 60 $CONSUL_URI
    sleep 2
fi
if ! [ -z "$REDIS_URI" ]; then
    wait-for -t 60 $REDIS_URI
    sleep 2
fi

# Interfaces and IPs
if [ -z "$INTERFACE_SIP" ]; then
    INTERFACE_SIP="eth0"
fi
if [ -z "$DISPATCHER_WEIGHT" ]; then
    DISPATCHER_WEIGHT="5"
fi
if [ "$DISPATCHER_WEIGHT" -gt "5" ]; then
    DISPATCHER_ORDER="primary"
else
    DISPATCHER_ORDER="secondary"
fi
if [ -z "$INTERFACE_DMQ" ]; then
    INTERFACE_DMQ="$INTERFACE_SIP"
fi
if [ -z "$INTERFACE_XHTTP" ]; then
    INTERFACE_XHTTP="$INTERFACE_SIP"
fi
SIP_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_SIP'" { gsub(/\/.*/, "", $4); print $4 }')
DMQ_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_DMQ'" { gsub(/\/.*/, "", $4); print $4 }')
XHTTP_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_XHTTP'" { gsub(/\/.*/, "", $4); print $4 }')

if [ -z "$XHTTP_PORT" ]; then
    XHTTP_PORT="9600"
fi

HOSTNAME=$(hostname)
export KAMAILIO=$(which kamailio)

# Kamailio-local.cfg
mkdir -p /etc/kamailio/ /etc/kamailio/dbtext

# IF ROUTER ROLE
echo '#!define ISNOT_SBC' > /etc/kamailio/kamailio-local.cfg
echo '#!define IS_ROUTER' > /etc/kamailio/kamailio-local.cfg

  #statements
echo '#!define LISTEN_XHTTP tcp:'$INTERFACE_XHTTP':'$XHTTP_PORT > /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_ROUTING_ENDPOINT "'$HTTP_API_ROUTING_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_CDR_ENDPOINT "'$HTTP_API_CDR_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_DBTEXT_UACREG_ENDPOINT "'$HTTP_API_DBTEXT_UACREG_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_TIMEOUT '$HTTP_API_TIMEOUT >> /etc/kamailio/kamailio-local.cfg
echo '#!define LISTEN '$LISTEN >> /etc/kamailio/kamailio-local.cfg
if ! [ -z "$TESTING" ]; then
    echo '#!define TESTING 1' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$DISPATCHER_ALG" ]; then
    echo '#!define DISPATCHER_ALG "'$DISPATCHER_ALG'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$HTTP_ASYNC_CLIENT_WORKERS" ]; then
    echo '#!define HTTP_ASYNC_CLIENT_WORKERS '$HTTP_ASYNC_CLIENT_WORKERS  >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$LISTEN_ADVERTISE" ]; then
    echo '#!define LISTEN_ADVERTISE '$LISTEN_ADVERTISE >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$ALIAS" ]; then
    echo '#!define ALIAS '$ALIAS >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$WITH_DMQ" ]; then
    echo '#!define WITH_DMQ 1' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_PORT "'$DMQ_PORT'"' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_LISTEN '$DMQ_LISTEN >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_SERVER_ADDRESS "sip:'$DMQ_IP':'$DMQ_PORT'"' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_NOTIFICATION_ADDRESS "'$DMQ_NOTIFICATION_ADDRESS'"' >> /etc/kamailio/kamailio-local.cfg
    if ! [ -z "$DMQ_PING_INTERVAL" ]; then
     echo '#!define DMQ_PING_INTERVAL "'$DMQ_PING_INTERVAL'"' >> /etc/kamailio/kamailio-local.cfg
    fi
fi
if ! [ -z "$WITH_REDIS_DIALOG" ]; then
    echo '#!define WITH_REDIS_DIALOG 1' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$DBURL_DIALOG" ]; then
    echo '#!define DBURL_DIALOG "'$DBURL_DIALOG'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$NAT_PING_FROM" ]; then
    echo '#!define NAT_PING_FROM "'$NAT_PING_FROM'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$ROUTER_AUTH_SECRET" ]; then
    echo '#!define ROUTER_AUTH_SECRET "'$ROUTER_AUTH_SECRET'"' >> /etc/kamailio/kamailio-local.cfg
fi

# test the config syntax
$KAMAILIO -f $KAMAILIO_CONF -c

# register/de-register service in consul
curl -i -X PUT http://${CONSUL_URI}/v1/agent/service/register -d '{
    "ID": "'$HOSTNAME'",
    "Name": "router",
    "Tags": ["router", "kamailio", "'$DISPATCHER_ORDER'"],
    "Address": "'$SIP_IP'",
    "Port": '$SIP_PORT',
    "Check": {
        "ID": "XHTTP",
        "Name": "XHTTP API on port 9600",
        "DeregisterCriticalServiceAfter": "10m",
        "Method": "GET",
        "HTTP": "http://'$XHTTP_IP':'$XHTTP_PORT'/status",
        "Timeout": "1s",
        "Interval": "10s"
    },
    "Weights": {
      "Passing": '$DISPATCHER_WEIGHT',
      "Warning": 1
    }
}'

exit_script() {
    curl -X PUT http://${CONSUL_URI}/v1/agent/service/deregister/$HOSTNAME
    [ -f /var/run/supervisor.sock ] && supervisorctl -c /etc/supervisor/conf.d/supervisord.conf shutdown
    date
    exit 143; # 128 + 15 -- SIGTERM
}
trap exit_script SIGINT SIGTERM

# run through supervisor
supervisord=$(which supervisord)
$supervisord -n -c /etc/supervisor/conf.d/supervisord.conf &

# wait for signals
while true; do sleep 1; done

# exit
exit_script


docker-entrypoint.sh
#!/bin/sh
date

# Determine SIP server type (SBC, ROUTER or AIO)
# AIO by default
if ! [ -z "$SIP_SERVER_TYPE" ]; then
    SIP_SERVER_TYPE='AIO'
fi

# Wait For
if ! [ -z "$ROUTER_CONFD_URI" ]; then
    wait-for -t 60 $ROUTER_CONFD_URI
fi
if ! [ -z "$CONSUL_URI" ]; then
    wait-for -t 60 $CONSUL_URI
    sleep 2
fi
if ! [ -z "$REDIS_URI" ]; then
    wait-for -t 60 $REDIS_URI
    sleep 2
fi

# Interfaces and IPs
if [ -z "$INTERFACE_SIP" ]; then
    INTERFACE_SIP="eth0"
fi
if [ -z "$DISPATCHER_WEIGHT" ]; then
    DISPATCHER_WEIGHT="5"
fi
if [ "$DISPATCHER_WEIGHT" -gt "5" ]; then
    DISPATCHER_ORDER="primary"
else
    DISPATCHER_ORDER="secondary"
fi
if [ -z "$INTERFACE_DMQ" ]; then
    INTERFACE_DMQ="$INTERFACE_SIP"
fi
if [ -z "$INTERFACE_XHTTP" ]; then
    INTERFACE_XHTTP="$INTERFACE_SIP"
fi
SIP_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_SIP'" { gsub(/\/.*/, "", $4); print $4 }')
DMQ_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_DMQ'" { gsub(/\/.*/, "", $4); print $4 }')
XHTTP_IP=$(ip -o -4 a | awk '$2 == "'$INTERFACE_XHTTP'" { gsub(/\/.*/, "", $4); print $4 }')

if [ -z "$XHTTP_PORT" ]; then
    XHTTP_PORT="9600"
fi

HOSTNAME=$(hostname)
export KAMAILIO=$(which kamailio)

# Kamailio-local.cfg
mkdir -p /etc/kamailio/ /etc/kamailio/dbtext
if [ "$SIP_SERVER_TYPE" = "AIO"]; then
    echo '#!define IS_AIO 1' > /etc/kamailio/kamailio-local.cfg
    echo '#!define ISNOT_SBC 1' > /etc/kamailio/kamailio-local.cfg
elif [ "$SIP_SERVER_TYPE" = "SBC"]; then
    echo '#!define IS_SBC 1' > /etc/kamailio/kamailio-local.cfg
elif [ "$SIP_SERVER_TYPE" = "ROUTER"]; then
    echo '#!define ISNOT_SBC 1' > /etc/kamailio/kamailio-local.cfg
    echo '#!define IS_ROUTER 1' > /etc/kamailio/kamailio-local.cfg
fi
  #statements
echo '#!define LISTEN_XHTTP tcp:'$INTERFACE_XHTTP':'$XHTTP_PORT > /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_ROUTING_ENDPOINT "'$HTTP_API_ROUTING_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_CDR_ENDPOINT "'$HTTP_API_CDR_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_DBTEXT_UACREG_ENDPOINT "'$HTTP_API_DBTEXT_UACREG_ENDPOINT'"' >> /etc/kamailio/kamailio-local.cfg
echo '#!define HTTP_API_TIMEOUT '$HTTP_API_TIMEOUT >> /etc/kamailio/kamailio-local.cfg
echo '#!define LISTEN '$LISTEN >> /etc/kamailio/kamailio-local.cfg
if ! [ -z "$TESTING" ]; then
    echo '#!define TESTING 1' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$HTTP_ASYNC_CLIENT_WORKERS" ]; then
    echo '#!define HTTP_ASYNC_CLIENT_WORKERS '$HTTP_ASYNC_CLIENT_WORKERS  >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$LISTEN_ADVERTISE" ]; then
    echo '#!define LISTEN_ADVERTISE '$LISTEN_ADVERTISE >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$ALIAS" ]; then
    echo '#!define ALIAS '$ALIAS >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$WITH_DMQ" ]; then
    echo '#!define WITH_DMQ 1' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_PORT "'$DMQ_PORT'"' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_LISTEN '$DMQ_LISTEN >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_SERVER_ADDRESS "sip:'$DMQ_IP':'$DMQ_PORT'"' >> /etc/kamailio/kamailio-local.cfg
    echo '#!define DMQ_NOTIFICATION_ADDRESS "'$DMQ_NOTIFICATION_ADDRESS'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$WITH_REDIS_DIALOG" ]; then
    echo '#!define WITH_REDIS_DIALOG 1' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$DBURL_DIALOG" ]; then
    echo '#!define DBURL_DIALOG "'$DBURL_DIALOG'"' >> /etc/kamailio/kamailio-local.cfg
fi
if ! [ -z "$ROUTER_AUTH_SECRET" ]; then
    echo '#!define ROUTER_AUTH_SECRET "'$ROUTER_AUTH_SECRET'"' >> /etc/kamailio/kamailio-local.cfg
fi

# test the config syntax
$KAMAILIO -f $KAMAILIO_CONF -c

# register/de-register service in consul
curl -i -X PUT http://${CONSUL_URI}/v1/agent/service/register -d '{
    "ID": "'$HOSTNAME'",
    "Name": "router",
    "Tags": ["router", "kamailio", "'$DISPATCHER_ORDER'"],
    "Address": "'$SIP_IP'",
    "Port": '$SIP_PORT',
    "Check": {
        "ID": "XHTTP",
        "Name": "XHTTP API on port 9600",
        "DeregisterCriticalServiceAfter": "10m",
        "Method": "GET",
        "HTTP": "http://'$XHTTP_IP':'$XHTTP_PORT'/status",
        "Timeout": "1s",
        "Interval": "10s"
    },
    "Weights": {
      "Passing": '$DISPATCHER_WEIGHT',
      "Warning": 1
    }
}'
exit_script() {
    curl -X PUT http://${CONSUL_URI}/v1/agent/service/deregister/$HOSTNAME
    [ -f /var/run/supervisor.sock ] && supervisorctl -c /etc/supervisor/conf.d/supervisord.conf shutdown
    date
    exit 143; # 128 + 15 -- SIGTERM
}
trap exit_script SIGINT SIGTERM

# run through supervisor
supervisord=$(which supervisord)
$supervisord -n -c /etc/supervisor/conf.d/supervisord.conf &

# wait for signals
while true; do sleep 1; done

# exit
exit_script
